### PR TITLE
[FRONT-125] Add 'Edit on GitHub' button to documentation pages

### PR DESCRIPTION
**Pull request by tumppi**
Uses `docsMap` to extract underlying file path for editing on GitHub and shows button when the path is available. This means that _Module Reference_ definitions are not currently editable.

### CODE REVIEW

**Review by fonty1 — APPROVED** [review]
Smooth! 😎

File: app/src/docs/components/DocsLayout/index.jsx [hunk 1]
```diff
@@ -1,42 +1,70 @@
...
-                    <div className={styles.content}>
-                        <BusLine dynamicScrollPosition>
-                            <MDXProvider components={Components}>
-                                <div {...props} />
-                            </MDXProvider>
-                        </BusLine>
-                        <PageTurner />
+const DocsLayout = ({ nav = <DocsNav />, location, staticContext, ...props }) => {
+    let editFilePath = null
+    Object.values(docsMap).some((doc) => {
```
Comment (juhah) [inline app/src/docs/components/DocsLayout/index.jsx:19]: This could be memoized, maybe

File: app/src/docs/components/DocsLayout/index.jsx [hunk 2]
```diff
@@ -1,42 +1,70 @@
...
+                        <div>
+                            <Navigation />
+                        </div>
+                        <div className={styles.content}>
+                            {editFilePath && (
+                                <Button
+                                    className={styles.editButton}
+                                    tag="a"
+                                    href={`https://github.com/streamr-dev/streamr-platform/edit/development/app/src/docs/content/${editFilePath}`}
+                                    kind="secondary"
```
Comment (juhah) [inline app/src/docs/components/DocsLayout/index.jsx:49]: Would it be better to open a new window from this button?

**Review by juhah — APPROVED** [review]
Grrreat 👍

File: app/src/docs/components/DocsLayout/index.jsx (hunk 1 above)
Comment (tumppi) [reply app/src/docs/components/DocsLayout/index.jsx:19]: True. I'll add 👍

File: app/src/docs/components/DocsLayout/index.jsx (hunk 2 above)
Comment (tumppi) [reply app/src/docs/components/DocsLayout/index.jsx:49]: Yeah, it would 👍

**Pull request merged**